assembly/onpremises-ide-packaging-war-factory/src/main/java/com/codenvy/onpremises/factory/filter/FactoryRetrieverFilter.java

```diff
@@ -109,6 +114,28 @@ public void doFilter(ServletRequest req, ServletResponse resp, FilterChain filte
                     return;
                 } 
                 requestedFactory = matchedFactories.get(0);
+            } else {
+                // asked for a parameters factory
+
+                // first populate map of parameters
+                Map<String, String> map = new HashMap<>();
+                Enumeration<String> parameterNames = httpReq.getParameterNames();
+                while (parameterNames.hasMoreElements()) {
+                    String parameterName  = parameterNames.nextElement();
+                    map.put(parameterName, httpReq.getParameter(parameterName));
+                }
+
+                // Create URL
+                final String getFactoryUrl = fromUri(apiEndPoint).path(ParametersFactoryService.class)
+                                                                 .path(ParametersFactoryService.class, "getFactory")
+                                                                 .build().toString();
+                // perform call
+                requestedFactory = httpRequestFactory.fromUrl(getFactoryUrl)
+                                                     .setMethod(POST)
+                                                     .setBody(map)
+                                                     .request()
+                                                     .asDto(Factory.class);
+
```

Comment:
done (Also don't need to check result of the call as it throws exception that are already catched)